index.js

```diff
@@ -32,7 +32,7 @@ function _credentials(options) {
     var baseUrl = _api_base_url(options) + '/v1/deployment/installer/agent/connectioninfo';
     debug('Trying to discover credentials from:', baseUrl);
 
-    var res = request('GET', baseUrl, { timeout: 5000, socketTimeout: 5000, headers: { Authorization: 'Api-Token ' + options.apitoken } });
+    var res = request('GET', baseUrl, { port: options.port, rejectUnauthorized: options.rejectUnauthorized, timeout: 5000, socketTimeout: 5000, headers: { Authorization: 'Api-Token ' + options.apitoken } });
```

Comment:
```suggestion
   const requestOptions = { port: options.port, rejectUnauthorized: options.rejectUnauthorized, ...(options.port != null) && port:options.port, ...(rejectUnauthorized != null) &&  rejectUnauthorized: options.rejectUnauthorized}
    var res = request('GET', baseUrl, requestOtions);
```

otherwise we would pass `port: undefined` which might be different to no port at all. Same for second field.